packages/service-library/src/data-providers/system-data-provider.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { CosmosContainerClient, cosmosContainerClientTypes } from 'azure-services';
+import { inject, injectable } from 'inversify';
+import { BatchPoolLoadSnapshot, ItemType, PartitionKey, ScanQueueLoadSnapshot } from 'storage-documents';
+
+export declare type BatchPoolAlias = 'urlScanPool';
+export declare type StorageQueueAlias = 'onDemandScanRequest';
+
+@injectable()
+export class SystemDataProvider {
```

Comment:
rename it to be more specific. may be TaskProcessingStatisticsDataProvider?

Reply:
ScanProcessingStateProvider